### DISCUSSION TITLE
ethers randomBytes() function and MaxUint256 constant equivalent

### DISCUSSION THREAD

**Discussion by Melancholism** [wevm/viem#667] [Question] [answered]
What is the [ethers randomBytes()](https://docs.ethers.org/v5/api/utils/bytes/#utils-randomBytes) function equivalent in viem?

I also need to access the [MaxUint256](https://docs.ethers.org/v5/api/utils/constants/#constants-MaxUint256) constant like ethers here in viem. But the only constant that I can access in viem is the `zeroAddress`. Can anyone help me, please?

Thanks in advance.

**Comment by jxom** [comment]
You can use native [`crypto.getRandomValues`](https://developer.mozilla.org/en-US/docs/Web/API/Crypto/getRandomValues) with a `Uint8Array`!

Can look into exporting constants like that. As for now, might be worth just manually adding them yourself.

**Comment by Melancholism** [comment]
Thanks for answering @jxom. 
Another function of crypto is [randomBytes()](https://nodejs.org/api/crypto.html#cryptorandombytessize-callback). Can I use it instead of `getRandomValues()`?
I have to mention that I need a random **_hex_** value. Below is what I am doing right now in my code:
```js
import { randomBytes } from 'crypto';

const randomHexValue = `0x${randomBytes(32).toString('hex')}`
```

And about the **_constants_**. I would really appreciate it if you could work on it and include them in **_viem_** like what **_ethers_** is doing. Also [here is another discussion](https://github.com/wagmi-dev/viem/discussions/447#discussion-5156932) that asked for the same feature. Anyway, I will manually add them for now and wait for future releases. Thanks a lot.

**Reply by jxom** [reply, marked as answer]
In the context of viem, bytes are represented as a `Uint8Array`, so `getRandomValues()` would probably fit a bit better. You can do something like:

```ts
import { bytesToHex } from 'viem'

const randomBytes = crypto.getRandomValues(new Uint8Array(32))
const randomHexValue = bytesToHex(randomBytes)
```

**Comment by xtools-at** [comment]
> And about the constants. I would really appreciate it if you could work on it and include them in viem like what ethers is doing.

i've added all missing ethers-v5 and -v6 constants in this PR: https://github.com/wagmi-dev/viem/pull/1475
